core/src/main/java/org/radargun/stages/stream/StreamStage.java

```diff
@@ -0,0 +1,214 @@
+import org.radargun.config.Property;
+import org.radargun.config.Stage;
+import org.radargun.reporting.Report;
+import org.radargun.stages.AbstractDistStage;
+import org.radargun.stages.cache.RandomDataStage;
+import org.radargun.state.SlaveState;
+import org.radargun.stats.DataOperationStats;
+import org.radargun.stats.DefaultStatistics;
+import org.radargun.stats.Statistics;
+import org.radargun.traits.*;
+import org.radargun.utils.Projections;
+import org.radargun.utils.TimeService;
+import org.radargun.utils.Utils;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Roman Macor &lt;rmacor@redhat.com&gt;
+ */
+@Stage(doc = "Stage which executes a specified stream Task against all keys in the cache.")
+public class StreamStage extends AbstractDistStage {
+
+   public static final String STREAM_RESULT_KEY = "streamResult";
+
+   @Property(optional = true, doc = "Boolean value that determines if the "
+           + "final results of the stream task are stored in the cache. "
+           + "The result object will be stored at key STREAM_RESULT_KEY. The default is false.")
+   private boolean storeResultInCache = false;
+
+   @Property(optional = true, doc = "Boolean value that determines if the "
+           + "final results of the stream are written to the log of the "
+           + "first slave node. The default is false.")
+   private boolean printResult = false;
+
+   @Property(doc = "The number of times to execute the stream task. The default is 10.")
+   private int numExecutions = 10;
+
+   @Property(doc = "Name of the cache where stream task should be executed. Default is the default cache.")
+   private String cacheName;
+
+   @Property(doc = "The name of the key in the MasterState object that returns the total number of "
+           + "bytes processed by the stream task. The default is RandomDataStage.RANDOMDATA_TOTALBYTES_KEY.")
+   private String totalBytesKey = RandomDataStage.RANDOMDATA_TOTALBYTES_KEY;
+
+   @Property(optional = false, doc = "Fully qualified class name of the StreamConsumer implementation.")
+   private String streamOperationClass;
+
+   @InjectTrait
+   private Streamable streamable;
+
+   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   private Clustered clustered;
+
+   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   private CacheInformation cacheInformation;
+
+   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   private BasicOperations basicOperations;
+
+   private Streamable.StreamFunction function;
+
+   private Object streamTaskResult;
+
+   @Override
+   public StageResult processAckOnMaster(List<DistStageAck> acks) {
+      StageResult result = super.processAckOnMaster(acks);
+      if (result.isError())
+         return result;
+
+      Report report = masterState.getReport();
+      Report.Test test = report.createTest("Stream_Stage", null, true);
```

Comment:
Please make the test name property (`testName` to be consistent with other stages)